tests/Carbon/PhpBug72338Test.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Carbon package.
+ *
+ * (c) Igor Deyashkin <igor_deyawka@mail.ru>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tests\Carbon;
+
+use Carbon\Carbon;
+use Tests\AbstractTestCase;
+
+/**
+ * The problem is, that $date->setTimezone($tz) with $tz in 'HH:MM' notation (["timezone_type"]=>int(1)) put DateTime object
+ *   on inconsistent state. It looks like internal timestamp becomes changed and it affects to such functions:
+ *
+ * * $date->modify() uses changed timestamp and result is wrong
+ *
+ * * $date->setTimezone($tz) settle this changed timestamp, even in case if $tz is not in 'HH:MM' format
+ *
+ * * $date->format('U') returns changed timestamp
+ *
+ * @link https://bugs.php.net/bug.php?id=72338 This bug on bugs.php.net
+ *
+ * @internal I use days changing in tests because using seconds|minute|hours may run setTimezone within.
+ */
+class PhpBug72338Test extends AbstractTestCase
+{
+    protected function setUp()
+    {
+        parent::setUp();
+
+        // In versions <= 5.5.5 setTimezone doesn't accept timezone in 'HH:MM' format.
+        // I dont know about versions between 5.5.5 and 5.5.18, but 5.5.18 accept it.
+        if (version_compare(PHP_VERSION, '5.5.18', '<')) {
+            $this->markTestSkipped();
+        }
+    }
+
+    /**
+     * Ensures that modify dont use changed timestamp
+     */
+    public function testModify()
+    {
+        $date = Carbon::createFromTimestamp(0);
+        $date->setTimezone('+02:00');
+        $date->modify('+1 day');
+        $this->assertSame('86400', $date->format('U'));
```

Comment:
If you want we can discuss all this things on Discord (it have web-interface, so you dont need ti install anything): https://discord.gg/FDbay